application/analyst/services/state_updater.py

```
@@ -328,13 +349,64 @@ def _update_knowledge(
+                # 简单情绪锚点：优先取末段中的情绪词
+                tail = paragraphs[-1] if paragraphs else content[-220:]
+                emotion_keywords = [
+                    "愤怒", "恐惧", "悲伤", "震惊", "警惕", "冷静", "苦涩", "决绝",
+                    "紧张", "困惑", "清醒", "痛苦", "压抑", "释然", "迟疑", "愧疚",
+                ]
+                hits = [kw for kw in emotion_keywords if kw in tail]
+                ending_emotion = "、".join(hits[:3]) if hits else tail[:60]
+
+                question_candidates = []
+                for chunk in reversed(paragraphs[-3:] if paragraphs else [tail]):
+                    parts = [seg.strip() for seg in chunk.replace("！", "？").split("？") if seg.strip()]
```

Ruff (0.15.10)

[warning] String contains ambiguous `！` (FULLWIDTH EXCLAMATION MARK). Did you mean `!` (EXCLAMATION MARK)?

(RUF001)